drivers/ipm/ipm_esp32.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_esp32_ipm
+#include "soc/dport_reg.h"
+#include "soc/gpio_periph.h"
+
+#include <stdint.h>
+#include <string.h>
+#include <device.h>
+#include <init.h>
+#include <drivers/ipm.h>
+#include <drivers/interrupt_controller/intc_esp32.h>
+#include <soc.h>
+#include <sys/atomic.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ipm_esp32, CONFIG_IPM_LOG_LEVEL);
+
+#define ESP32_IPM_LOCK_FREE_VAL 0xB33FFFFF
+
+__packed struct esp32_ipm_control {
+	uint8_t dest_cpu_msg_id[4];
+	atomic_val_t lock;
+};
+
+__packed struct esp32_ipm_memory {
+	uint8_t wifi_shm[CONFIG_ESP32_IPM_CHANNEL_WIFI_SIZE];
+	uint8_t bt_shm[CONFIG_ESP32_IPM_CHANNEL_BT_SIZE];
+	uint8_t user_0_shm[CONFIG_ESP32_IPM_CHANNEL_USER_0_SIZE];
+	uint8_t user_1_shm[CONFIG_ESP32_IPM_CHANNEL_USER_1_SIZE];
+};
+
+struct esp32_ipm_data {
+	ipm_callback_t	cb[5];
+	uint32_t this_core_id;
+	uint32_t other_core_id;
+	struct esp32_ipm_memory *shm; 
+	struct esp32_ipm_control *control;
+};
+
+static struct esp32_ipm_data esp32_ipm_device_data;
+
+static size_t esp32_ipm_channel_data_size(uint32_t id) {
+	switch(id) {
```

Comment:
Please check what is the commit you're looking at, this is not appearing into the latest commit.

Probably after I squash the commits this will be more clear